src/max_number_occurence.py

```diff
@@ -9,20 +9,22 @@ def generated_list():
     return random_generated_list
 
 
-def generated_dictionary():
-    numbers_occurensces = {}
-    random_generated_list = generated_list()
+def generated_dictionary(random_list):
```

Comment:
this function can work with any list, not just random list - so name `random_list` is not very correct. you can call it more generally `numbers_list`.